### DISCUSSION TITLE
How to exit from FeedHold state

### DISCUSSION THREAD

**Discussion by vilianch123** [arkypita/LaserGRBL#1610] [General]
Hello,arkypita,
I have tested grbl with FeedHold function.When I touched Feed Hold button
in lasergrbl UI,the moving would stopped.But I have no idea how to exit from
this FeedHold state.
I have seen from Mach4 UI,which have both Cycle Start button and Feed Hold button,
and if we want exit from FeedHold state,we could touch Cycle Start button again?
Do lasergrbl has the same logic processing?
Thank you!
![Mach4 FeedHold](https://user-images.githubusercontent.com/86226681/136789399-0afef9d2-551b-471d-9e99-b464a651e23a.PNG)

**Comment by arkypita** [comment]
The button near feed hold is the resume button, it doesn't seem complicated.
https://lasergrbl.com/usage/user-interface/

**Reply by vilianch123** [reply]
Hi arkypita,
But for my program,it seems the RESUME button can not work,it was always gray and when I touched it,nothing resume!
Please see attachment!
![Resume button can not work](https://user-images.githubusercontent.com/86226681/136961572-a7c410f8-d9d0-4df9-8be6-fea897509161.PNG)

**Reply by arkypita** [reply]
It is greyed out because it is not in the state of "Feed Hold" but in state of "Run", so there is nothing to resume.
This is how the "Feed Hold" state look like:

![image](https://user-images.githubusercontent.com/8782035/136963176-a1eebf66-f5bf-4a2c-a587-fdd8e4e52c17.png)

**Comment by arkypita** [comment]
When you say:

> I have tested grbl with FeedHold function.When I touched Feed Hold button
> in lasergrbl UI,the moving would stopped.But I have no idea how to exit from
> this FeedHold state.

you mean that when you press the "red hand" button in bottom-right, the machine stop but doesn't go to hold state and the resume (orange button) is not enabled?

**Reply by vilianch123** [reply]
Do you mean Resume should work in "hold" state?
Perhaps I got it. I have not changed into hold state correctly, although moving stopped.
The program,you know,was ported into my own hardware plat and the original grbl firmware has been changed.
So I would try it.
Thank you.

**Reply by arkypita** [reply]
Yes, LaserGRBL UI is developed to match GRBL spec. UI react with machine state (also because the state of the machine can change on its own, not necessarily because LaserGRBL commands it to change) In detail, the code that conditions the UI buttons is as follows: 
https://github.com/arkypita/LaserGRBL/blob/7da08dfec4a582f077845cfb2db6f0912faca59f/LaserGRBL/Core/GrblCore.cs#L2227
